src/Service/Factory/AdapterManagerFactory.php

```diff
@@ -80,6 +80,9 @@ public function createService(ServiceLocatorInterface $serviceLocator)
 
         $serviceConfig = new Config($serviceConfig);
 
-        return new AdapterManager($serviceConfig);
+        $adapterManager = new AdapterManager($serviceConfig);
+        $adapterManager->setServiceLocator($serviceLocator);
```

Comment:
`$serviceLocator` should be set in constructor, but if we need to support php 5.4 I have to set it like this :D